mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/quota/CassandraPerUserMaxQuotaManagerMigrationTest.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.james.core.quota.QuotaSizeLimit;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.domainlist.api.mock.SimpleDomainList;
+import org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils;
+import org.apache.james.mailbox.cassandra.modules.CassandraQuotaModule;
+import org.apache.james.mailbox.model.QuotaRoot;
+import org.apache.james.mailbox.quota.MaxQuotaManager;
+import org.apache.james.mailbox.quota.UserQuotaRootResolver;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.cassandra.CassandraUsersDAO;
+import org.apache.james.user.cassandra.CassandraUsersRepositoryModule;
+import org.apache.james.user.lib.UsersRepositoryImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mockito;
+
+import com.google.inject.Injector;
+
+public class CassandraPerUserMaxQuotaManagerMigrationTest {
+
+    private static final Domain DOMAIN = Domain.of("domain.tld");
+    private static final Username USERNAME = Username.of("bob");
+    private static final QuotaRoot QUOTA_ROOT = QuotaRoot.quotaRoot("#private&bob", Optional.empty());
+
+    @RegisterExtension
+    static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(CassandraModule.aggregateModules(
+        CassandraUsersRepositoryModule.MODULE,
+        CassandraQuotaModule.MODULE,
+        CassandraMutualizedQuotaModule.MODULE));
+
+    private static CassandraPerUserMaxQuotaManagerMigration migration;
+    private static MaxQuotaManager oldMaxQuotaManager;
+    private static MaxQuotaManager newMaxQuotaManager;
+    private static UsersRepository usersRepository;
+
+    @BeforeAll
+    static void setup() throws Exception {
+        DomainList domainList = new SimpleDomainList();
+        domainList.addDomain(DOMAIN);
+        CassandraUsersDAO usersDAO = new CassandraUsersDAO(cassandraCluster.getCassandraCluster().getConf());
+        usersRepository = new UsersRepositoryImpl<>(domainList, usersDAO);
+        Injector testInjector = GuiceUtils.testInjector(cassandraCluster.getCassandraCluster());
+        oldMaxQuotaManager = testInjector.getInstance(CassandraPerUserMaxQuotaManagerV1.class);
+        newMaxQuotaManager = testInjector.getInstance(CassandraPerUserMaxQuotaManagerV2.class);
+        UserQuotaRootResolver userQuotaRootResolver = Mockito.mock(UserQuotaRootResolver.class);
+        Mockito.when(userQuotaRootResolver.forUser(USERNAME)).thenReturn(QUOTA_ROOT);
+        migration = new CassandraPerUserMaxQuotaManagerMigration(usersRepository,
+            domainList,
+            oldMaxQuotaManager,
+            newMaxQuotaManager,
+            userQuotaRootResolver);
+    }
+
+    @Test
+    void shouldMigrateGlobalLimit() throws Exception {
+        oldMaxQuotaManager.setGlobalMaxMessage(QuotaCountLimit.count(100));
+        oldMaxQuotaManager.setGlobalMaxStorage(QuotaSizeLimit.size(100));
```

Comment:
Could you use different values for message and storage? To enforce everythin is right. Also Those values are all Long, add the "L" at the end of each value

For all tests